Automattic/jetpack
# Search: Fix chosen filter options not visible when no results returned

Fixes #14179 and partially fixes `pcNPJE-av-p2`

#### Changes proposed in this Pull Request:
The PR proposed a solution where the frontend cache the Aggregations object, and then use it to to still show the filters when there's no result. When the overlay opens, the cache would be cleared.

The patch only works when there's no result and there are filters in the URL, so it wouldn't affect normal search flow.

I considered to put the filter label in the URL, but which could potentially cause XSS issue or show unwanted content on the page.

Another solution would be mapping the filters on the backend, which basically requires a re-write of a few frontend parts in PHP, not very ideal considering the PR is just a workaround.

The ultimate solution would be changes from the Search API side, and return filters including counts on all requests, which would need support from elastic search side.

The PR does NOT fix the issue when empty search results is shared by URL, because I think this situation is relatively rare, because if there's no results, people wouldn't tend to share it.

#### Jetpack product discussion
pcNPJE-av-p2

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Go to a site with Jetpack subscription
- Search something and then check one/more filter options
- Continue to enter the query until there's no results
- Ensure the filters are still visible for users to uncheck
- Ensure the filters work as expected

![Screen Capture on 2021-08-16 at 18-40-54](https://user-images.githubusercontent.com/1425433/129521762-14d09db0-d791-45cb-ae24-5ada8748d338.gif)
